File: 04-05/func_while.py
def func_while():
    r = input("Você quer continuar? (s/n)")
    if r == "s" or r == "S":
        print("Você continuou...insistente!")
        func_while()
    else:
        print("Você desistiu...finalmente!")

if __name__ == "__main__":
    func_while()

File: 04-05/funcWhile.hs
funcWhile = do
    putStrLn "Você quer continuar? (s/n)"
    r <- getLine
    if r == "s" || r == "S" then do
        putStrLn "Você continuou...insistente!"
        funcWhile
    else 
        putStrLn "Você desistiu...finalmente!"

main :: IO ()
main = do
    funcWhile